Sources/FoundationEssentials/FileManager/FileManager+Files.swift

```diff
@@ -553,49 +553,68 @@ extension _FileManagerImpl {
     func attributesOfItem(atPath path: String) throws -> [FileAttributeKey : Any] {
 #if os(Windows)
         return try path.withNTPathRepresentation { pwszPath in
-            var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
-            guard GetFileAttributesExW(pwszPath, GetFileExInfoStandard, &faAttributes) else {
-                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
-            }
-
             let hFile = CreateFileW(pwszPath, GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, nil, OPEN_EXISTING, 0, nil)
             if hFile == INVALID_HANDLE_VALUE {
                 throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
             }
             defer { CloseHandle(hFile) }
 
+            var info: BY_HANDLE_FILE_INFORMATION = BY_HANDLE_FILE_INFORMATION()
+            guard GetFileInformationByHandle(hFile, &info) else {
+              throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+
             let dwFileType = GetFileType(hFile)
             let fatType: FileAttributeType = switch (dwFileType) {
                 case FILE_TYPE_CHAR: FileAttributeType.typeCharacterSpecial
                 case FILE_TYPE_DISK:
-                    faAttributes.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY
+                    info.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY
                             ? FileAttributeType.typeDirectory
                             : FileAttributeType.typeRegular
                 case FILE_TYPE_PIPE: FileAttributeType.typeSocket
                 case FILE_TYPE_UNKNOWN: FileAttributeType.typeUnknown
                 default: FileAttributeType.typeUnknown
             }
 
-            let size: UInt64 = (UInt64(faAttributes.nFileSizeHigh) << 32) | UInt64(faAttributes.nFileSizeLow)
-            let creation: Date = Date(timeIntervalSince1970: faAttributes.ftCreationTime.timeIntervalSince1970)
-            let modification: Date = Date(timeIntervalSince1970: faAttributes.ftLastWriteTime.timeIntervalSince1970)
-            return [
+            let systemNumber = UInt64(info.dwVolumeSerialNumber)
+            let systemFileNumber = UInt64(info.nFileIndexHigh << 32) | UInt64(info.nFileIndexLow)
+            let referenceCount = UInt64(info.nNumberOfLinks)
+
+            let isReadOnly = info.dwFileAttributes & FILE_ATTRIBUTE_READONLY != 0
+            // Directories are always considered executable, but we check for other types
+            let isExecutable = fatType == .typeDirectory || SaferiIsExecutableFileType(pwszPath, 0)
```

Comment:
```suggestion
            let isExecutable = fatType == .typeDirectory || SaferiIsExecutableFileType(pwszPath, false)
```

Reply:
Unfortunately that didn't compile for me:

```
C:\Users\jmschonfeld\Documents\Repos\swift-foundation\Sources\FoundationEssentials\FileManager\FileManager+Files.swift:587:54: error: cannot convert value of type 'Bool' to expected argument type 'BOOLEAN' (aka 'UInt8')
585 |                 true // Directories are always considered executable
586 |             } else {
587 |                 SaferiIsExecutableFileType(pwszPath, false)
    |                                                      `- error: cannot convert value of type 'Bool' to expected argument type 'BOOLEAN' (aka 'UInt8')
588 |             }
589 |             var posixPermissions = UInt16(_S_IREAD)
```

It looks like `BOOLEAN` is `UInt8` and not expressible by a boolean literal?

Reply:
That is good to know. We can deal with that later if we want.